modules/repository_base/repository.tf

```diff
@@ -11,36 +10,48 @@ locals {
 resource "github_repository" "repository" {
   name        = var.name
   description = var.description
-  visibility  = var.visibility
+  #trivy:ignore:avd-git-0001
+  visibility = var.visibility
 
-  auto_init              = true
-  archive_on_destroy     = false
-  has_downloads          = var.has_downloads
-  has_issues             = var.has_issues
-  has_projects           = var.has_projects
-  has_wiki               = var.has_wiki
-  has_discussions        = var.has_discussions
-  vulnerability_alerts   = var.has_vulnerability_alerts
-  topics                 = var.topics
-  homepage_url           = var.homepage
-  delete_branch_on_merge = var.delete_head_on_merge
-  allow_auto_merge       = var.allow_auto_merge
+  auto_init                   = true
+  archive_on_destroy          = false
+  has_downloads               = var.has_downloads
+  has_issues                  = var.has_issues
+  has_projects                = var.has_projects
+  has_wiki                    = var.has_wiki
+  has_discussions             = var.has_discussions
+  vulnerability_alerts        = var.has_vulnerability_alerts
+  topics                      = var.topics
+  homepage_url                = var.homepage
+  delete_branch_on_merge      = var.delete_head_on_merge
+  allow_auto_merge            = var.allow_auto_merge
   web_commit_signoff_required = var.requires_web_commit_signing
-  license_template       = var.license_template
+  license_template            = var.license_template
 
 
   # A hacky way of getting around the 422 errors received from github api
   dynamic "security_and_analysis" {
     for_each = local.can_configure_security_and_analysis ? [1] : []
     content {
-      advanced_security {
-        status = var.advance_security ? "enabled" : "disabled"
+      dynamic "advanced_security" {
+        for_each = var.advance_security != null ? [var.advance_security] : []
+        content {
+          status = var.advance_security ? "enabled" : "disabled"
+        }
```

Comment:
I don't think they can ever be null. I was thinking of something more like
```suggestion
      dynamic "advanced_security" {
        for_each = var.advance_security ? [var.advance_security] : []
        content {
          status = "enabled"
        }
```
The reason being that the github provider has been running into 422 errors and one of those errors was when someone attempts to set advance security on a repo who's owner is not a github enterprise account. Even if you set it to disabled it still seems to throw the 422. It might have been fixed in the newest version of the provider but I'm not sure.